swagger.yaml

```diff
@@ -0,0 +1,321 @@
+      - $ref: '#/parameters/id'
+      - $ref: '#/parameters/edition'
+      - $ref: '#/parameters/version'
+      - $ref: '#/parameters/dimension'
+      responses:
+        200:
+          description: "Json object containing all values for a dimension"
+          schema:
+            $ref: '#/definitions/DimensionValues'
+        404:
+          description: "No dataset edition with the specific version was found using the id, edition and version provided or the dimension does not exist within the dataset"
+        500:
+          description: "Failed to process the request due to a internal error"
+definitions:
+  Dataset:
+    type: object
+    description: "An object containing information about published datasets from the ONS"
+    properties:
+      id:
+        type: integer
+        description: "The dataset id"
+      title:
+        type: string
+        description: "The title for the dataset"
+      url:
+        type: string
+        description: "An url to the dataset"
+      release_date:
+        type: string
+        description: "The release date of this dataset"
+      next_release:
+        type: string
+        description: "The next release date for this dataset"
+      edition:
+        type: string
+        description: "The edition of the dataset"
+      version:
+        type: string
+        description: "The version of the dataset"
+      contact:
+        type: object
+        description: "Object containing contact information for this dataset"
+        properties:
+          name:
+            type: string
+            description: "The name of the person"
+          telephone:
+            type: string
+            description: "Telephone number to contact the person"
+          email:
+            type: string
+            description: "An email address to contact the person"
+  Datasets:
+    type: object
+    properties:
+      items:
+        type: array
+        description: "An array of Datasets"
+        items:
+          $ref: '#/definitions/Dataset'
+      number_of_results:
+        type: integer
+        description: "The number of results available"
+  Dimension:
+    type: object
+    description: "A single dimension within a dataset"
+    properties:
+      codeListId:
+        type: string
+        description: "The id of the code list that refers to this dimension"
+      id:
+        type: string
+        description: "The id of the dimension"
+      name:
+        type: string
+        description: "The name of the dimension"
+      type:
+        type: string
+        description: "The type of dimension"
+  Dimensions:
+    type: object
+    properties:
+      items:
+        type: array
+        description: "An array of dimensions"
+        items:
+          $ref: '#/definitions/Dimension'
+      number_of_results:
+              type: integer
+              description: "The number of results available"
+  DimensionValues:
+    type: object
+    properties:
+      code_list_id:
```

Comment:
Dimension values might not always come from the same code list (a geography could be compiled of multiple code lists). Maybe DimensionValues should follow the pattern of Dimensions and be an array, and each single Dimension Value can reference its specific code list, or rather link to a specific value inside a code list - which would contain the metadata Matt is talking about in his comment.